db/mongo/mongodb.go

```diff
@@ -96,36 +99,45 @@ func (m *mongoClient) RemoveTask(taskID string) error {
 
 // ListTasks removes the tasks from mongo db
 func (m *mongoClient) ListTasks() ([]*models.Task, error) {
+	var task *models.Task
+
 	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(taskCollection)
-	// Create an empty slice to store the retrieved tasks
-	tasks := []*models.Task{}
 
-	// Create an empty filter since we want to retrieve all tasks
+	tasks := []*models.Task{}
 	filter := bson.M{}
-	// Execute the find operation on the collection
-	cursor, err := collection.Find(context.Background(), filter)
+
+	cursor, _ := collection.Find(context.Background(), filter)
+	err := collection.FindOne(context.Background(), filter).Decode(&task)
+	if err == mongo.ErrNoDocuments {
+		return nil, domainErr.NewAPIError(domainErr.NotFound, fmt.Sprintf("task: %s not found"))
+	}
+
```

Comment:
why both Find and FindOne are used?

`
taskList := make([]*models.Task, 0)
collection := m.conn.Database(viper.GetString(config.DbName)).Collection(taskCollection)
filter := bson.M{}

cursor, err := collection.Find(ctx, filter)
	if err != nil {
		return nil, errors.Wrap(err, "failed to retrieve list of tasks")
	}
	defer cursor.Close(ctx)
`
this should be fine. please test it this way if this works for you